.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,38 @@
+# Contributing Guidelines for TerseTS
+
+The repository is a Zig project with C and Python bindings. It uses the Apache‑2.0 license and includes a pre‑commit hook for building, formatting, and running tests. The following guidelines summarize how to contribute effectively.
+
+## 1. Environment Setup 
+* Install a recent Zig toolchain (v0.14.0 is used in the GitHub workflow).
+* For Python bindings, install `setuptools` (and `build` for distribution) as shown in the Python README.
+* Build the project in `Debug` mode with `zig build` and in Release modes with `zig build -Doptimize=ReleaseFast`, etc., as outlined in the README.
+
+## 2. Running Checks
+* Use the provided tools/pre-commit script before committing. It builds the library, runs tests, checks formatting, and compiles the C header across Debug and ReleaseFast profiles.
+* `zig fmt --check .` is run to enforce formatting; ensure the code is formatted accordingly.
+
+## 3. Code Style
+* Every source file begins with the Apache‑2.0 license header. Copy the header from existing files, such as `src/functional/poor_mans_compression.zig`
+* Module descriptions use `//!` comments and public functions use `///` comments. Follow this pattern in new Zig files.
+* Keep functions concise and document parameters and returned values.
+* Follow as close as possible [Zig coding style](https://ziglang.org/documentation/master/#Style-Guide).
+* Every source file containing a new compression method must follow this convention:
+  * `enum` and `struct` without methods are first.
+  * The `compress` and `decompress` functions come right after, respectivelly.
+  * `struct` containing methods after `decompress` function. 
+  * Private functions follow in the same order in which they are called.
+  * All test cases at the end of the file.
+
+## 4. Tests
```

Comment:
Again I think it is better to split this into section or README files per language so each section can contain the information needed for a user of a specific language that wants to contribute. My assumption is that a user who wants to contribute mainly uses a specific language and is thus focused on getting started contributing using a specific language. If we have a binding specific contribution file per language, then this main contributors file can focus on the big picture, e.g., how to add a new compression method with steps and links to each of other READMEs when relevant.